### PR TITLE
Bug 1227593 - Sync engine shared lib. r=michielbdejong

### PR DESCRIPTION
**Pull request by ferjm**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=192523cebf717e5ccfe604ef1688cf01cf0ab482

File: build/eslint/xfail.list [hunk 1]
```diff
@@ -233,6 +233,7 @@ shared/js/intl/l20n-client.js # line 43, col 11, Error - Unsafe assignment to in
 shared/js/lazy_loader.js # line 55, col 11, Error - Unsafe assignment to innerHTML (no-unsafe-innerhtml/no-unsafe-innerhtml)
 shared/js/smart-screen/shared_utils.js # line 29, col 7, Error - Unsafe assignment to innerHTML (no-unsafe-innerhtml/no-unsafe-innerhtml)
 shared/js/template.js # line 160, col 9, Error - Unsafe assignment to innerHTML (no-unsafe-innerhtml/no-unsafe-innerhtml)
+shared/js/sync/ext/kinto.min.js # line 217, col 1044, Error - Using postMessage() with wildcard targets is not allowed. (no-wildcard-postmessage/no-wildcard-postmessage)
```
Comment (michielbdejong) [inline build/eslint/xfail.list]: I guess you could manually remove https://github.com/mozilla-b2g/gaia/pull/33439/files#diff-3098656e507a6e322b7c045f4c782506R298 but no harm in leaving it in there.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2aace6b9c7d1d22e451a9d21d51f16de62568bdb

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=936c8cb75b930746bf53417af279e9360edf2bf1

**Pull request merged**